packages/palette_generator/lib/palette_generator.dart

```diff
@@ -309,10 +307,10 @@ class PaletteGenerator with Diagnosticable {
     return maxScoreSwatch;
   }
 
-  PaletteColor _getMaxScoredSwatchForTarget(
+  PaletteColor? _getMaxScoredSwatchForTarget(
       PaletteTarget target, Set<Color> usedColors) {
     double maxScore = 0.0;
-    PaletteColor maxScoreSwatch;
+    PaletteColor? maxScoreSwatch;
```

Comment:
This can be `late`, and the `if (maxScoreSwatch == null ||`... can be removed below.

Reply:
Tests failing if it is late:

```dart
dart:_internal                                           LateError._throwLocalNotInitialized
package:palette_generator/palette_generator.dart 322:12  PaletteGenerator._getMaxScoredSwatchForTarget
package:palette_generator/palette_generator.dart 300:9   PaletteGenerator._generateScoredTarget
package:palette_generator/palette_generator.dart 290:11  PaletteGenerator._selectSwatches
package:palette_generator/palette_generator.dart 77:5    new PaletteGenerator.fromColors
package:palette_generator/palette_generator.dart 124:29  PaletteGenerator.fromImage

LateInitializationError: Local 'maxScoreSwatch' has not been initialized.

```

Reply:
Oh, yeah, I guess it can't be late, if `_shouldBeScoredForTarget` always returns false.